homeassistant/components/modbus/sensor.py

```diff
@@ -28,71 +21,27 @@
 from homeassistant.helpers.restore_state import RestoreEntity
 from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
 
-from . import number
 from .base_platform import BasePlatform
 from .const import (
-    CALL_TYPE_REGISTER_HOLDING,
-    CALL_TYPE_REGISTER_INPUT,
     CONF_DATA_TYPE,
-    CONF_HUB,
-    CONF_INPUT_TYPE,
     CONF_PRECISION,
-    CONF_REGISTER,
-    CONF_REGISTER_TYPE,
-    CONF_REGISTERS,
-    CONF_REVERSE_ORDER,
     CONF_SCALE,
     CONF_SWAP,
     CONF_SWAP_BYTE,
-    CONF_SWAP_NONE,
     CONF_SWAP_WORD,
     CONF_SWAP_WORD_BYTE,
-    DATA_TYPE_CUSTOM,
-    DATA_TYPE_FLOAT,
-    DATA_TYPE_INT,
     DATA_TYPE_STRING,
-    DATA_TYPE_UINT,
-    DEFAULT_HUB,
-    DEFAULT_SCAN_INTERVAL,
-    DEFAULT_STRUCT_FORMAT,
     MODBUS_DOMAIN,
 )
+from .modbus import ModbusHub
 
 PARALLEL_UPDATES = 1
 _LOGGER = logging.getLogger(__name__)
 
 
-PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
-    {
-        vol.Required(CONF_REGISTERS): [
-            {
-                vol.Required(CONF_NAME): cv.string,
-                vol.Required(CONF_REGISTER): cv.positive_int,
-                vol.Optional(CONF_COUNT, default=1): cv.positive_int,
-                vol.Optional(CONF_DATA_TYPE, default=DATA_TYPE_INT): vol.In(
-                    [
-                        DATA_TYPE_INT,
-                        DATA_TYPE_UINT,
-                        DATA_TYPE_FLOAT,
-                        DATA_TYPE_STRING,
-                        DATA_TYPE_CUSTOM,
-                    ]
-                ),
-                vol.Optional(CONF_DEVICE_CLASS): DEVICE_CLASSES_SCHEMA,
-                vol.Optional(CONF_HUB, default=DEFAULT_HUB): cv.string,
-                vol.Optional(CONF_OFFSET, default=0): number,
-                vol.Optional(CONF_PRECISION, default=0): cv.positive_int,
-                vol.Optional(
-                    CONF_REGISTER_TYPE, default=CALL_TYPE_REGISTER_HOLDING
-                ): vol.In([CALL_TYPE_REGISTER_HOLDING, CALL_TYPE_REGISTER_INPUT]),
-                vol.Optional(CONF_REVERSE_ORDER, default=False): cv.boolean,
-                vol.Optional(CONF_SCALE, default=1): number,
-                vol.Optional(CONF_SLAVE): cv.positive_int,
-                vol.Optional(CONF_STRUCTURE): cv.string,
-                vol.Optional(CONF_UNIT_OF_MEASUREMENT): cv.string,
-            }
-        ]
-    }
+PLATFORM_SCHEMA = vol.All(
+    # Deprecated in Home Assistant 2021.7
+    cv.deprecated(SENSOR_DOMAIN),
```

Comment:
This schema isn't correct. `cv.deprecated` doesn't work well with platform schemas since there's no single key specific for this platform. Just use the existing warning instead.

Reply:
I was my recommendation after I had seen it used on entire CONFIG_SCHEMA, but good to know.

This PR is for 2021.7